js/Utility/Models.js

```diff
@@ -0,0 +1,132 @@
+/*
+* adapt-contrib-trickle
+* License - http://github.com/adaptlearning/adapt_framework/LICENSE
+* Maintainers - Oliver Foster <oliver.foster@kineo.com>
+*/
+
+define([
+    'coreJS/adapt',
+    '../DataTypes/StructureType'
+], function(Adapt, StructureType) {
+
+    var ModelUtilities = {
+
+        getDescendantsFlattened: function(id, parentFirst) {
```

Comment:
could use a brief comment explaining bit more about what this does and why/when it is needed